### ISSUE TITLE
QR codes are unchecked

### ISSUE THREAD

**Issue by fritzalder** [railslove/rcvr-app#68]
Hi,
just saw this app and found a problem with how you handle QR codes.

### The issue
If you deploy this in the wild and on tables in the real world, you really need to check the URL that barcodes are pointing to. Otherwise you can easily run Man in the middle attacks with forged URLs that gather the peoples' data while still checking them in. Or do worse things than that with an unchecked QR code. 

If I see this correctly, this line is to blame: https://github.com/railslove/rcvr-app/blob/b3b918915953f37a830110c3604961fd6600ebc2/pages/qr.tsx#L25

### How do I reproduce it?
Just use any QR code generator to produce a proper URL and scan it from your app. It will redirect the browser directly there without any warnings or messages to the user.

### How to fix this?
Do proper input vetting (this is hard!) and do a double check that the QR code does not contain anything malicious. Another approach to this problem could be to not just encode a dumb link in the QR code but instead something like: <qr_code_id, RCVR_signature> where you additionally provide a signature that signs the code id as valid. Only if the signature checks out for the given ID and is signed by you, the app uses the code and appends it to the rcvr.app URL.

**Comment by fritzalder** [comment]
I totally agree that a lot of users will use their own QR-code scanner. So the easiest option is of course to just remove this feature.

At the same time however, if you provide this feature on your website, then you should also make sure that it is at least as safe as any QR code scanner out there. And nowadays, most scanners asks the user to confirm the scanner URL. I think no one is expecting you to solve the usability problem of potentially unsafe QR codes ;) but a simple check whether the scanned target conforms to expected parameters would already be a good first step.

**Comment by salimhb** [comment]
I would implement a check in our own QR-code scanner (https://rcvr.app/qr), as you suggest. I wouldn't however completely drop the feature allowing guest to use their default camera app, it's the most friction free way to scan QR-code nowadays.
